connect/api/src/main/java/org/apache/kafka/connect/sink/SinkTask.java

```diff
@@ -25,9 +25,34 @@
 import java.util.Map;
 
 /**
- * SinkTask is a Task takes records loaded from Kafka and sends them to another system. In
- * addition to the basic {@link #put} interface, SinkTasks must also implement {@link #flush}
- * to support offset commits.
+ * SinkTask is a Task that takes records loaded from Kafka and sends them to another system. Each task
+ * instance is assigned a set of partitions by the Connect framework and will handle all records received
+ * from those partitions. As records are fetched, they will be written to the downstream system using
+ * {@link #put(Collection)}. At a minimum, you must implement this interface along with {@link #flush(Map)},
+ * which is used to commit the offsets of records forwarded. Generally, after records have been flushed,
```

Comment:
This is a bit misleading since `flush` itself doesn't do that, it's just part of the offset commit process. Not a major issue, but this could probably be a bit clearer.

The following sentence also seems a bit confusing. Since I know how offset commit works, I assume you mean they will only be reprocessed if there is a failure between the `flush()` and the subsequent offset commit, but unless offset commit is explained somewhere not every developer will know these details.

Not sure if we should expound upon this here, make a reference somewhere else (e.g., I don't think we have any package level docs yet...) or assume that for full details they'd reference the developer guide.